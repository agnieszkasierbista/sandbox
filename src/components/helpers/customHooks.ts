import React from 'react';

export function useScrollBlocking(isModalVisible: boolean) {

    React.useEffect(() => {
        if (isModalVisible) {
            document.body.style.overflow = "hidden"
        } else {
            document.body.style.overflow = "initial"
        }
    }, [isModalVisible])

}