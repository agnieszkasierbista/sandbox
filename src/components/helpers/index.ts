export * from './helpers';
export * from './formatters';
export * from './validators';