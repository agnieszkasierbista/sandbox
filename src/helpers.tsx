import { Background, Blur, CancelButton, Color, ConfirmButton, Creator, FormFieldNameAndValueArray, XButton } from "./App.types";

export function findEmptyFields(data: FormFieldNameAndValueArray): string[] {

    const emptyFields = data.filter(field => {
        const fieldValue = field[1];

        if (typeof fieldValue === "string") {
            return fieldValue.length === 0;
        }
    });

    return emptyFields.map(emptyField => emptyField[0]);

}
export function isString(data: FormFieldNameAndValueArray): boolean {

    const isAString = data.find(entry => {
        const entryValue = entry[1];

        if (typeof entryValue === "string") {
            return entryValue.length > 0;
        }
    });

    return !!isAString;
}

// export function getConfirmButton(props: any) {
//     switch (props.confirmButton[1]) {
//         case "confirm":
//             return props.dispatchConfirm()
//         case "save":
//             return props.dispatchSave()
//         case "confirmAndClose":
//             return props.dispatchConfirmAndClose()
//         default:
//             return console.log("Confirm button clicked!")
//     }
// }

export function getConfirmButton(confirmButton: ConfirmButton, option1: () => void, option2: () => void, option3: () => void) {
    switch (confirmButton[1]) {
        case "confirm":
            return option1()
        case "save":
            return option2()
        case "confirmAndClose":
            return option3()
        default:
            return console.log("Confirm button clicked!")
    }
}

// export function getCancelButton(props: any) {
//     switch (props.cancelButton[1]) {
//         case "cancel":
//             return props.dispatchCancel()
//         case "clear":
//             return props.dispatchClear()
//         case "cancelAndClose":
//             return props.dispatchCancelAndClose()
//         default:
//             return console.log("Cancel button clicked!")
//     }
// }

export function getCancelButton(cancelButton: CancelButton, option1: () => void, option2: () => void, option3: () => void) {
    switch (cancelButton[1]) {
        case "cancel":
            return option1()
        case "clear":
            return option2()
        case "cancelAndClose":
            return option3()
        default:
            return console.log("Cancel button clicked!")
    }
}

// export function getXbutton(props: any) {
//     switch (props.xButton[1]) {
//         case "close":
//             return props.dispatchClose()
//         case "cancelX":
//             return props.dispatchCancelX()
//         case "confirmX":
//             return props.dispatchConfirmX()
//         default:
//             return console.log("X button clicked!")
//     }
// }

export function getXbutton(xButton: XButton, option1: () => void, option2: () => void, option3: () => void) {
    switch (xButton[1]) {
        case "close":
            return option1()
        case "cancelX":
            return option2()
        case "confirmX":
            return option3()
        default:
            return console.log("X button clicked!")
    }
}
