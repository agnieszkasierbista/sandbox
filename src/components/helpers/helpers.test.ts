import {findEmptyFields, isString} from "./helpers"

test('finds empty strings in touples and returns array of touple[0] for them', () => {
    expect(findEmptyFields([["1", ""], ["2", "aaa"], ["3", "bbb"], ["4", ""], ["5", "ccc"]])).toEqual(["1", "4"]);
});

test('takes array of touples and evaluates if the second value of a touple is a string', () => {
    expect(isString([["1", ""], ["2", "aaa"], ["3", "bbb"], ["4", ""], ["5", "ccc"]])).not.toBe(true);
});

test("asdf", () => { expect("tre").not.toContain("tr")});

function newFunc(name?: string) {
    if(name) {
        return {
            name: name
        }
    }
    return {}
}

describe("newFunc", () => {
    it("newFunc", () => {
        expect(newFunc()).toEqual({})
    });
    xit("newFunc", () => {
        expect(newFunc("abc")).toEqual({name: "abc"})
    });
    xit("newFunc", () => {
        expect(newFunc("xyz")).toEqual({name: "xyz"})
    });
    test.each(["abc", "xyz"])('test with %s entry',  (entry) => {
        expect(newFunc(entry)).toEqual({name: entry})
    });
});




