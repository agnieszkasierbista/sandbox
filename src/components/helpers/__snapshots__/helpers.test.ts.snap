// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`newFunc newFuncSnap 1`] = `
Object {
  "name": "abc",
}
`;
